src/comment.rs

```diff
@@ -1001,24 +1019,29 @@ fn light_rewrite_comment(
     config: &Config,
     is_doc_comment: bool,
 ) -> String {
-    let lines: Vec<&str> = orig
+    let lines: Vec<String> = orig
```

Comment:
Do you think this could be `Vec<Cow>` instead of `Vec<String>`? It feels like we only need to add a space in the case where we've got a multi-line block comment to keep it properly aligned.

Reply:
I don't understand rust well enough to understand whether `Cow` is better.  However, in any case, `Vec<Cow>` gives compilation error, and either `Vec<Cow<`_, str>>` or `Vec<Cow<`_, String>>` produce the following error message:

>  .collect();
     ^^^^^^^ value of type `Vec<Cow<'_, str>>` cannot be built from `std::iter::Iterator<Item=std::string::String>`

I am not sure how to resolve this error.

Reply:
Here's another idea. We could reimplement `light_rewrite_comment` altogether as follows: 
```rust
/// Trims whitespace and aligns to indent, but otherwise does not change comments.
fn light_rewrite_comment(
    orig: &str,
    offset: Indent,
    config: &Config,
    is_doc_comment: bool,
) -> String {
    let mut result = String::with_capacity(orig.len() * 2);
    let mut lines = orig.lines().peekable();

    // This is basically just l.trim(), but in the case that a line starts with `*` we want to leave
    // one space before it, so it aligns with the `*` in `/*`.
    while let Some(line) = lines.next() {
        let first_non_whitespace = line.find(|c| !char::is_whitespace(c));
        if let Some(fnw) = first_non_whitespace {
            if line.as_bytes()[fnw] == b'*' {
                result.push(' ');
            }
            // Preserve markdown's double-space line break syntax in doc comment.
            let trimmed = trim_end_unless_two_whitespaces(&line[fnw..], is_doc_comment);
            result.push_str(trimmed);
        };

        let is_last = lines.peek().is_none();
        if !is_last {
            result.push_str(&offset.to_string_with_newline(config))
        }
    }
    result
}
```
 In the new implementation we wouldn't need to collect the lines into a `Vec<_>` at all. Instead, we'd just push the trimmed lines to a pre allocated buffer as we go along, and in the event that we get to a `*` line we just prepend a single space before pushing the trimmed line.

Reply:
Changed to the new suggested implementation.  Thanks.